### PR TITLE
content/assignments: Add "Lambda Function Loader" hackathon

### PR DESCRIPTION
**Pull request by teodutu**
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [x] Tested your changes against relevant architectures and platforms;
- [x] Updated relevant documentation (if needed).

## Description of changes

- Add the hackathon description
- Add grading and submission instructions
- Update `config.yaml` to render the new changes

### CODE REVIEW

File: content/assignments/lambda-function-loader/README.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# Lambda Function Loader
+
+## Application Development
+
+All development work should be carried out exclusively within the [virtual machines dedicated to Operating Systems](https://cs-pub-ro.github.io/operating-systems/resources#virtual-machine).
+
+**Note:**
+It's crucial to avoid running tests locally, either on personal computers or within your allocated virtual machines.
+Discrepancies might arise between local environments and the dedicated OS virtual machines.
+Grading will consider results obtained solely from the OS virtual machine environment.
```
Comment (Alex-deVis) [inline content/assignments/lambda-function-loader/README.md:10]:
```suggestion
Grading will consider results obtained solely in the OS virtual machine environment.
```

File: content/assignments/lambda-function-loader/README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+The core of this project revolves around the construction of a system capable of dynamic library loading and subsequent function execution.
+The project architecture embraces a client-server model, where the server is responsible for receiving and fulfilling requests to execute functions sourced from specific dynamic libraries present within the system.
```
Comment (Alex-deVis) [inline content/assignments/lambda-function-loader/README.md:13]:
```suggestion
The project architecture embraces a [client-server model](https://cs-pub-ro.github.io/operating-systems/Lab/IO/client-server-model), where the server is responsible for receiving and fulfilling requests to execute functions sourced from specific dynamic libraries present within the system.
```

File: content/assignments/lambda-function-loader/README.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+- Prepare/pre-hooks: Initial operations independent of loading the library or executing functions.
+- Library loading: Loading the library and other related management tasks.
+- Execution of requested functionality:
+Execution of the specified function from the library.
+If `funcname` is absent, the `run` function is invoked.
+The `filename` parameter is permissible only when `funcname` is specified.
+- Unloading the library: Procedures associated with unloading the library from memory.
+- Post-hooks:
+Operations conducted after execution, regardless of the library's presence in memory.
+**Tip:**
```
Comment (Alex-deVis) [inline content/assignments/lambda-function-loader/README.md:54]:
```suggestion
Operations conducted after execution, regardless of the library's presence in memory.

**Tip:**
```

**Review by Alex-deVis — APPROVED** [review]

File: content/misc/rules-and-grading.md [hunk 4]
```diff
@@ -156,6 +156,73 @@ If you miss a lab or do not receive the maximum score on other labs, the lab bon
...
+  - team composition
+  - GitLab link
+  - implementation details;
+  each extra functionality requires a description, approach, and testing method.
+
+#### Awards
+
+Each member of the winning teams will receive a prize based on their ranking:
+
+1. TODO: prizes
```
Comment (elena19m) [inline content/misc/rules-and-grading.md:216]:
```suggestion
1. First place: 900 RON gift card.
2. Second place: 800 RON gift card.
3. Third place: 700 RON gift card.
```

File: content/misc/rules-and-grading.md [hunk 5]
```diff
@@ -156,6 +156,73 @@ If you miss a lab or do not receive the maximum score on other labs, the lab bon
...
+
+#### Equivalence
+
+All participating teams are **eligible** for one point equivalence in the OS subject (depending on **implementation complexity and project stage developed during the hackathon**).
+
+#### Winners Announcement
+
+Winning teams will be announced by TODO-date.
```
Comment (elena19m) [inline content/misc/rules-and-grading.md:224]:
```suggestion
Winning teams will be announced by December 20, 2023.
```

**Review by fabianpatras — APPROVED** [review]

**Review by elena19m — APPROVED** [review]

File: content/assignments/lambda-function-loader/src/server.c [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	return lib_posthooks(lib);
+}
+
+static int parse_command(const char *buf, char *name, char *func, char *params)
+{
+	int ret;
+
+	ret = sscanf(buf, "%s %s %s", name, func, params);
+	if (ret != 3)
+		return -1;
```
Comment (elena19m) [inline content/assignments/lambda-function-loader/src/server.c:78]: These two lines should be deleted.  `sscanf` returns the number of items (strings) assigned. If `ret == 1`, we should run the default function (`run`) from the `name` library; if `ret == 2`, we should run the `func` function from the `name` library etc.

**Review by elena19m — CHANGES_REQUESTED** [review]

**Comment by teodutu** [comment]
Done @elena19m. I also updated the solution in cs-pub-ro/operating-systems-internal#13 accordingly.

**Pull request merged**